tiledb/sm/tile/tile.h

```diff
@@ -77,6 +77,16 @@ class TileBase {
   /*                API                */
   /* ********************************* */
 
+  /**
+   * Set the tile data type. Used in filter pipeline for filters that modify
+   * tile type. If the filter output datatype is ANY, the datatype is unchanged.
+   *
+   * @param datatype The updated Datatype of the tile.
+   */
+  inline void set_datatype(Datatype datatype) {
+    type_ = datatype == Datatype::ANY ? type_ : datatype;
```

Comment:
This logic seems not awesome since its accounting for filter internals in a different class. Without that context it doesn't seem to make much sense in not actually applying a setter behavior for what appears to be a setter from the outside.

Reply:
Yeah, I was on the fence about this. I moved this check up to FilterPipeline code and only set the datatype when needed, so this will behave as a setter in all cases.